bin/pycbc_upload_xml_to_gracedb

```diff
@@ -153,22 +198,44 @@ for event in coinc_table:
     xmldoc.childNodes[-1].appendChild(coinc_inspiral_table_curr)
     xmldoc.childNodes[-1].appendChild(coinc_event_map_table_curr)
     xmldoc.childNodes[-1].appendChild(sngl_inspiral_table_curr)
-    ligolw_utils.write_filename(xmldoc, "tmp_coinc_xml_file.xml")
+
+    if args.filename_T050017:
+        ifos = sorted([sngl.ifo for sngl in sngl_inspiral_table_curr])
+        ifos_str = ''.join(ifos)
+        id_str = args.filename_T050017
+        filename_xml = "{}-PyCBC_{}-{:d}-1.xml".format(ifos_str, id_str, time)
+        filename_psd = "{}-PyCBC_{}-{:d}-1_psd.xml.gz".format(ifos_str, id_str, time)
+    else:
+        print("Using default filenames")
+        filename_xml = "tmp_coinc_xml_file.xml"
+        filename_psd = "tmp_psd.xml.gz"
+
+    ligolw_utils.write_filename(xmldoc, filename_xml)
     psd_xmldoc = lal.series.make_psd_xmldoc(psddict)
-    ligolw_utils.write_filename(psd_xmldoc, "tmp_psd.xml.gz", gz=True)
+    ligolw_utils.write_filename(psd_xmldoc, filename_psd, gz=True)
+
+    if args.no_upload:
+        # We have saved the file, and aren't uploading, so reset the tables
+        # and move to the next event
+        xmldoc.childNodes[-1].removeChild(coinc_event_table_curr)
+        xmldoc.childNodes[-1].removeChild(coinc_inspiral_table_curr)
+        xmldoc.childNodes[-1].removeChild(coinc_event_map_table_curr)
+        xmldoc.childNodes[-1].removeChild(sngl_inspiral_table_curr)
+        continue
+
     if args.testing:
-        r = gracedb.createEvent("Test", "pycbc", "tmp_coinc_xml_file.xml",
+        r = gracedb.createEvent("Test", "pycbc", filename_xml,
                                 search="AllSky", offline=True).json()
     else:
-        r = gracedb.createEvent("CBC", "pycbc", "tmp_coinc_xml_file.xml",
+        r = gracedb.createEvent("CBC", "pycbc", filename_xml,
```

Comment:
While you are at it, may I ask to reduce these several lines to something like this:
```python
group_tag = 'Test' if args.testing else 'CBC'
r = gracedb.createEvent(group_tag, 'pycbc', filename_xml…
```